contracts/ProtocolReserve/ProtocolShareReserve.sol

```diff
@@ -92,8 +260,37 @@ contract ProtocolShareReserve is Ownable2StepUpgradeable, ExponentialNoError, Re
      */
     function updateAssetsState(
         address comptroller,
-        address asset
-    ) public override(IProtocolShareReserve, ReserveHelpers) {
-        super.updateAssetsState(comptroller, asset);
+        address asset,
+        IncomeType incomeType
+    ) public override(IProtocolShareReserve) {
+        require(ComptrollerInterface(comptroller).isComptroller(), "ProtocolShareReserve: Comptroller address invalid");
+        require(asset != address(0), "ProtocolShareReserve: Asset address invalid");
+        require(
+            (comptroller != CORE_POOL_COMPTROLLER &&
+                PoolRegistryInterface(POOL_REGISTRY).getVTokenForAsset(comptroller, asset) != address(0)) ||
+                (comptroller == CORE_POOL_COMPTROLLER),
```

Comment:
With a new internal function (in this contract), that will iterate on `comptroller.allMarkets` looking for a market with the `asset` as underlying token. It could be expensive, in terms of gas cost, but I don't know any other option...